modules/platform/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocumentTest.java

```diff
@@ -129,4 +130,27 @@ public void testLifeCycleResetOnCopy() throws OperationException {
             assertEquals(initialLifeCycleState, result.getCurrentLifeCycleState());
         }
     }
+
+    @Test
+    public void testVersionNotNullOnCopy() throws OperationException {
+        DocumentModel source = session.createDocumentModel(ROOT, "Source", FILE);
+        source = session.createDocument(source);
+        source = session.getDocument(session.checkIn(source.getRef(), VersioningOption.MAJOR, ""));
+
+        assertEquals(1, (long) source.getPropertyValue("uid:major_version"));
+        assertEquals(0, (long) source.getPropertyValue("uid:minor_version"));
+
+        Map<String, Serializable> params = Map.of(TARGET_PROPERTY_KEY, ROOT, NAME_PROPERTY_KEY, COPY_DOC_NAME);
+
+        try (OperationContext context = new OperationContext(session)) {
+            context.setInput(source);
+            DocumentModel result = (DocumentModel) automationService.run(context, CopyDocument.ID, params);
+            txFeature.nextTransaction();
```

Comment:
This line is not needed since you're directly manipulating the output document of the Automation service call (not passing through the HTTP layer to execute the operation then re-fetching the document afterwards).